packages/flutter/lib/src/cupertino/spell_check_suggestions_toolbar.dart

```diff
@@ -0,0 +1,108 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show SuggestionSpan;
+import 'package:flutter/widgets.dart';
+
+import 'text_selection_toolbar.dart';
+import 'text_selection_toolbar_button.dart';
+
+/// iOS only shows 3 spell check suggestions in the toolbar.
+const int maxSuggestions = 3;
+
+/// The default spell check suggestions toolbar for Android.
+///
+/// Tries to position itself below the [anchors], but if it doesn't fit, then it
+/// readjusts to fit above bottom view insets.
+class CupertinoSpellCheckSuggestionsToolbar extends StatelessWidget {
+  /// Constructs a [SpellCheckSuggestionsToolbar].
+  const CupertinoSpellCheckSuggestionsToolbar({
+    super.key,
+    required this.anchors,
+    required this.buttonItems,
+  });
+
+  /// The location on which to anchor the menu.
+  final TextSelectionToolbarAnchors anchors;
+
+  /// The [ContextMenuButtonItem]s that will be turned into the correct button
+  /// widgets and displayed in the spell check suggestions toolbar.
+  ///
+  /// See also:
+  ///
+  ///  * [AdaptiveTextSelectionToolbar.buttonItems], the list of
+  ///    [ContextMenuButtonItem]s that are used to build the buttons of the
+  ///    text selection toolbar.
+  ///  * [SpellCheckSuggestionsToolbar.buttonItems], the list of
+  ///    [ContextMenuButtonItem]s used to build the Material style spell check
+  ///    suggestions toolbar.
+  final List<ContextMenuButtonItem> buttonItems;
+
+  /// Builds the button items for the toolbar based on the available
+  /// spell check suggestions.
+  static List<ContextMenuButtonItem>? buildButtonItems(
+    BuildContext context,
+    EditableTextState editableTextState,
+  ) {
+    // Determine if composing region is misspelled.
+    final SuggestionSpan? spanAtCursorIndex =
+      editableTextState.findSuggestionSpanAtCursorIndex(
+        editableTextState.currentTextEditingValue.selection.baseOffset,
+    );
+
+    if (spanAtCursorIndex == null) {
+      return null;
+    }
+    if (spanAtCursorIndex.suggestions.isEmpty) {
+      return <ContextMenuButtonItem>[
+        ContextMenuButtonItem(
+          onPressed: () {},
+          label: 'No Replacements Found',
+        )
+      ];
+    }
+
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    // Build suggestion buttons.
+    int suggestionCount = 0;
+    for (final String suggestion in spanAtCursorIndex.suggestions) {
+      if (suggestionCount < maxSuggestions) {
```

Comment:
Maybe break out early:

```dart
if (suggestionCount >= maxSuggestions) {
  break;
}
```

That way, in the odd case where there were thousands of suggestions or something, you wouldn't iterate through all of them.